docs/installation/database/passing-configuration.rst

```diff
@@ -0,0 +1,430 @@
+environment is used.  If no ``default`` environment is known, an error is
+raised.  It is strongly recommended that a ``default`` environment be defined
+in all configuration files - either explicitly, or as an alias to an explicitly
+defined environment.
+
+If no environment named ``default`` is known, but one named ``datacube`` **IS**
+known, the ``datacube`` environment is used and a deprecation warning issued.
+``datacube`` will be dropped as a legacy default environment name in a future
+release.
+
+4. Generic Environment Variable Overrides
+-----------------------------------------
+
+Configuration values in config files can be over-ridden by setting the appropriate environment variable.
+
+The name of overriding environment variables are all upper-case and structured:
+
+.. code-block:: bash
+
+   $ODC_{environment name}_{option name}
+
+E.g. to override the :confval:`db_password` field in the ``main`` environment,
+set the ``$ODC_MAIN_DB_PASSWORD`` environment variable.
+
+Environment variables overrides are **NOT** applied to environments defined in
+configuration that was passed in explicitly as a string or dictionary.
+
+4a. Dynamic Environments
+++++++++++++++++++++++++
+
+It is possible for environments to be defined dynamically purely in environment variables.
+
+E.g. given the following active configuration file:
+
+.. code-block::yaml
+
+     default:
+         alias: main
+     main:
+         index_driver: postgres
+         db_url: postgresql://myuser:mypassword@server.domain/main
+
+and the following defined environment variables:
+
+.. code-block::bash
+
+   ODC_AUX_INDEX_DRIVER=postgis
+   ODC_AUX_DB_URL=postgres://auxuser:secret@backup.domain/aux
+
+You can request the "aux" environment and it's configuration will be
+dynamically read from the environment variables, even though it is not
+mentioned in the configuration file at all.
+
+Notes:
+
+1. Environment variables are read when configuration is first read from that
+   environment (i.e. when first connecting to the database.)
+
+2. As all configuration options have default values, it is no longer possible
+   to get an error by requesting an environment that does not exist.  Instead,
+   an all-defaults environment with the requested name will be dynamically
+   created.  The only exception is when a specific environment is not
+   requested.  In this case, the ``default`` environment is only used if it is
+   either defined in the active configuration file or has previously been
+   explicitly requested from the same :py:class:`ODCConfig` object.
+
+3. Although environment variable overrides are bypassed for configured
+   environments by passing in explicit configuration, reading from environment
+   variables to dynamically create new environments is still supported.
+
+4b. Environment Variable Overrides and Environment Aliases
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To avoid troublesome and unpredictable corner cases, aliases can only be
```

Comment:
I think the "To avoid troublesome and unpredictable corner cases," can be omitted.